.shippable.yml

```diff
@@ -15,7 +12,16 @@ build:
     - export CROSS_COMPILE64="ccache aarch64-linux-gnu-"
     - export CFG_DEBUG_INFO=n
     - export CFG_WERROR=y
-    - function _make() { make -j$(getconf _NPROCESSORS_ONLN) -s O=out $* && ccache -s && ccache -z; }
+    - export START=$(date +%s)
+    - export PROJ=$ORG_NAME-$REPO_NAME
+    - export SCP_OPT="-o ConnectTimeout=10 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+
+    - function download_cache() { ssh $SCP_OPT shippable@shippable-cache.forissier.org "cat ccache-$PROJ.tar.gz" | tar zx -C /root || echo Nevermind; }
+    - function upload_cache() { if [ ! -e .uploaded ]; then echo Uploading cache && tar c -C /root .ccache | gzip -1 | ssh $SCP_OPT shippable@shippable-cache.forissier.org "cat >ccache-$PROJ.tar.gz" && touch .uploaded || echo Nevermind; fi; }
```

Comment:
The `cat >ccache-$PROJ.tar.gz` will nuke the already present cache when starting to upload the new cache. So if there's some error before the new cache is completely uploaded we'll start from scratch the next time.
I don't know if this is a problem in practice though.

Reply:
Right, not a problem in practice, if that happens and if the build does time out due to the cache missing, we'll just restart the build and hope the cache upload won't fail this time.